app/controllers/view_components_system_test_controller.rb

```diff
@@ -1,30 +1,27 @@
 # frozen_string_literal: true
 
-class ViewComponentsSystemTestController < ActionController::Base # :nodoc:
-  before_action :validate_test_env
-  before_action :validate_file_path
+if Rails.env.test?
```

Comment:
I can't think of a reason we'd need this outside of the test env, but I wonder if we should gate it here, or gate the `require` statement?

Then folks could opt-in to requiring it if they did need it outside the test env for whatever reason. but like I said, I don't know that there's a valid use-case for that.

Reply:
Given the existing logic for raising if outside of the test env, I think we're safe here. I also don't see us loading this with `require`?

Reply:
This change seems reasonable but it breaks in production – where Rails uses `eager_load`. See #2269.